exporters/sender/okhttp/src/main/java/io/opentelemetry/exporter/sender/okhttp/internal/OkHttpHttpSender.java

```diff
@@ -145,6 +145,8 @@ public byte[] responseBody() throws IOException {
                                   return body.bytes();
                                 }
                               });
+                        } catch (Throwable e) {
+                          onError.accept(e);
```

Comment:
I'm not sure I understand this. Calling `response.body()` doesn't (shouldn't?) throw. Its calling body.bytes() that throws and that's only going to do so when the caller invoked `response.responseBody()`.

Reply:
Yes, removed this and added some tests for coverage on the marshaling